main.go

```diff
@@ -486,13 +511,37 @@ func handleSigterm(cancel func()) {
 	cancel()
 }
 
-func serveMetrics(address string) {
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, _ *http.Request) {
+func registerMetrics(mx *http.ServeMux) {
+	mx.HandleFunc("/healthz", func(w http.ResponseWriter, _ *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		w.Write([]byte("OK"))
 	})
 
-	http.Handle("/metrics", promhttp.Handler())
+	mx.Handle("/metrics", promhttp.Handler())
+}
+
+func loadCertificates(mx *http.Server, config *externaldns.Config) error {
+	tlsConfig := &tls.Config{}
+
+	caCert, err := os.ReadFile(config.TLSServerCA)
+	if err != nil {
+		return err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig.ClientCAs = caCertPool
+	tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
```

Comment:
From the docs:
```
	// RequireAndVerifyClientCert indicates that a client certificate should be requested
	// during the handshake, and that at least one valid certificate is required
	// to be sent by the client.
	RequireAndVerifyClientCert
```

That means every client is expected to authenticate via certificate. TLS support for metrics and health endpoints is already a somewhat unusual request and requiring all clients to authenticate via certificate is even more surprising.

I don't think this logic belongs in external-dns. Putting a reverse proxy in front of external-dns is certainly a viable solution and could be used here instead.